tests/openedx_tagging/core/tagging/test_api.py

```diff
@@ -578,72 +651,33 @@ def test_autocomplete_tags(self, search: str, expected_values: list[str], expect
                 _value=value,
             ).save()
 
-        # Test for open taxonomy
-        self._validate_autocomplete_tags(
-            open_taxonomy,
-            search,
-            expected_values,
-            [None] * len(expected_ids),
-        )
+        result = tagging_api.search_tags(closed_taxonomy, search)
+        assert pretty_format_tags(result, parent=False) == expected
 
-        # Test for closed taxonomy
-        self._validate_autocomplete_tags(
-            closed_taxonomy,
-            search,
-            expected_values,
-            expected_ids,
-        )
-
-    def test_autocompleate_not_implemented(self) -> None:
-        with self.assertRaises(NotImplementedError):
-            tagging_api.autocomplete_tags(self.taxonomy, 'test', None, object_tags_only=False)
+    def test_autocomplete_tags_closed_omit_object(self) -> None:
+        """
+        Test autocomplete search that omits the tags from a specific object
+        """
+        object_id = 'new_object_id'
+        tagging_api.tag_object(object_id=object_id, taxonomy=self.taxonomy, tags=["Archaebacteria"])
+        result = tagging_api.search_tags(self.taxonomy, "ChA", exclude_object_id=object_id)
+        assert pretty_format_tags(result, parent=False) == [
+            "Archaea (used: 0, children: 3)",
+            "  Euryarchaeida (used: 0, children: 0)",
+            "  Proteoarchaeota (used: 0, children: 0)",
+            # These results are no longer included because of exclude_object_id:
+            # "Bacteria (used: 0, children: 2)",  # does not contain "cha" but a child does
+            # "  Archaebacteria (used: 1, children: 0)",
+        ]
 
-    def _get_tag_values(self, tags) -> list[str]:
+    def _get_tag_values(self, tags: TagDataQuerySet) -> list[str]:
         """
         Get tag values from tagging_api.autocomplete_tags() result
         """
-        return [tag.get("value") for tag in tags]
+        return [tag["value"] for tag in tags]
 
     def _get_tag_ids(self, tags) -> list[int]:
         """
         Get tag ids from tagging_api.autocomplete_tags() result
         """
         return [tag.get("tag_id") for tag in tags]
```

Comment:
Nit: `_get_tag_values()` and `_get_tag_ids()` are no longer used, so can be removed.